confirmation/models.py

```diff
@@ -77,6 +78,13 @@ def get_object_from_key(
     obj = confirmation.content_object
     assert obj is not None
 
+    used_value = confirmation_settings.STATUS_USED
+    revoked_value = confirmation_settings.STATUS_REVOKED
+    if hasattr(obj, "status") and obj.status in [used_value, revoked_value]:
+        # Confirmations where the object has the status attribute are one-time use
+        # and are marked after being used (or revoked).
+        raise ConfirmationKeyException(ConfirmationKeyException.EXPIRED)
```

Comment:
Would it be cleaner to refactor this to just call something like `confirmation.can_be_used()` to contain this fiddly logic in a method of the `Confirmation` class?

Reply:
I'm not sure -- the `mark_object_used` code path accesses those already, which makes me feel like such an extraction would not clearly help.